### DISCUSSION TITLE
Propagate authenticated LDAP credentials to Kafka Connect

### DISCUSSION THREAD

**Discussion by ritambanik** [tchiotludo/akhq#1407] [Q&A] [answered]
Hello,

I want to configure access to Connect clusters from AKHQ.  Below is what I have configured,

> connections:

       xyz-kafka-dev:
           connect:
               - name: "kafka-connect-dev-kc-1"
                  url: "http:/kafka-connect-dev-1.xyz-corp.cloud:8083"
                  basic-auth-username: "user1"
                  basic-auth-password: "password"

But my Kafka Connect cluster is RBAC enabled and I would like to pass the authenticated LDAP credentials (used to login to AKHQ) to Kafka Connect instead of the above hardcoded credentials. This way each user can have it's own view of the connectors list.

Please let me know how can I achieve this?

Thanks!

**Comment by tchiotludo** [comment, marked as answer]
It's not possible for now, I've open an issues https://github.com/tchiotludo/akhq/issues/1409 
PR are welcome

**Reply by ritambanik** [reply]
Thanks for clarifying.

**Reply by ritambanik** [reply]
Hello,

Can you please guide me on where in the code the static credentials are being passed to the Schema Registry API calls? Our Schema Registry is RBAC enabled and if we can pass the contextual user, the API calls would return filtered results or block unauthorized calls. I would like to try adding that to the AKHQ Schema Registry flow as without this feature we are not able to use AKHQ in our production environment.

Thanks!

**Reply by AlexisSouquiere** [reply]
Here (and especially line 298) at the SR client creation:

https://github.com/tchiotludo/akhq/blob/cf173a61e241ff6dd6c9bd14819c7487e216fdf6/src/main/java/org/akhq/modules/KafkaModule.java#L294-L300

**Reply by ritambanik** [reply]
Thanks for the reply. Can you please let me know the below,

1. Is KafkaModule instantiated per user login?
2. Is there a way to get the logged in user principal from the KafkaModule?
